Sources/xcproj/PBXBuildRule.swift

```diff
@@ -0,0 +1,136 @@
+import Foundation
+
+/// A PBXBuildRule is used to specify a method for transforming an input file in to an output file(s).
+final public class PBXBuildRule: PBXObject, Equatable {
+
+    // MARK: - Attributes
+
+    /// Element compiler spec.
+    public var compilerSpec: String
+
+    /// Element file patterns.
+    public var filePatterns: String?
+
+    /// Element file type.
+    public var fileType: String
+
+    /// Element is editable.
+    public var isEditable: Bool
+
+    /// Element name.
+    public var name: String?
+
+    /// Element output files.
+    public var outputFiles: [String]
+
+    /// Element output files compiler flags.
+    public var outputFilesCompilerFlags: [String]?
+
+    /// Element script.
+    public var script: String?
+
+    // MARK: - Init
+
+    public init(reference: String,
+                compilerSpec: String,
+                filePatterns: String? = nil,
+                fileType: String,
```

Comment:
One more very minor thing. As `fileType` is non optional (I don't know if it should be or not), can we move it before optional `filePatterns`?